_posts/2019-06-16-good-first-issue.md

```diff
@@ -0,0 +1,116 @@
+
+The problem I had with this approach is when you subscribe to the repo, you get notification for not only ALL the issues reported, but also new release, commits, PR’s, wiki updates etc. which can get overwhelming if you want to subscribe to more than one project.
+
+
+### So what is goodfirstissue app
+
+```goodfirstissue``` is a github app, which once installed on repo/org, will notify first time contributors through [@goodfirstissue](https://twitter.com/goodfirstissue) handle. It tweets only when there is an unassigned and open issue which also have a label ```good first issue```. This way you can filter out all the other noise that you get when you subscribe to repo/org directly.
+
+it consists of two parts:
+
+* [A github app](https://github.com/apps/goodfirstissue) that is installed on org/repo who encourage first time contributors to contribute to their projects. This sends notification to a openfaas function which filters noise and tweet the issue if it qualifies for a ```good first issue```
+* [A twitter handle](https://twitter.com/goodfirstissue) that is followed by potential ‘first time contributors’ who can now get real time notification about ```good first issue``` from awesome opensource projects in a variety of different languages.
+
+### Architecture
+
+<img src="/images/good-first-issue/architecture.png" />
+
+### Why function and not a restful webservice
+
+The first version of goodfirstissue app was deployed as a webservice on Kubernetes cluster on Digital Ocean. Deployment was easy, but few things were bothering me:
+
+* Manual configuration of my custom domain referencing to loadbalancer created by DO.
+* Getting ```https``` endpoint for the service. using https gives more confidence to your users.
+* Infrastructure Cost. Once promotional credits were over it was going to cost money to run this cluster. The maintenance of this cluster was also going to cost me time.
+
+Although above could be achieved relatively easily with tools like [cert-manager](https://github.com/jetstack/cert-manager/), but I didn't wanted to get into managing the Kubernetes cluster myself.
+
+### Enter OpenFaaS and its community cluster
+
+While I was collecting feedback on this function, Alex suggested me to change this app to a function and also made me aware of OpenFaaS community cluster which provided me:
+
+* CI/CD with git pushes
+* Free hosting
+* Free https endpoint
+* Secure Secrets management using Sealed Secrets
+* Easy debugging using slack events/bot commands. (this is so awesome !!)
+* Easy access to dashboard with pre-created metrics about the function.
+* Access to OpenFaaS community to share my experience with them, and learn from their experiences.
+
+### Current Users
+
```

Comment:
Maybe quote one or two of the high-profile tweets.

Who were your first users and how have they benefited? Did they give feedback or tweet about the bot? If not, can you ask them to do so.. then we can embed it here?

Reply:
done